cylc/flow/cycling/iso8601.py

```diff
@@ -897,13 +897,13 @@ def _interval_parse(interval_string):
     return WorkflowSpecifics.interval_parser.parse(interval_string)
 
 
-def point_parse(point_string):
+def point_parse(point_string: str) -> 'TimePoint':
     """Parse a point_string into a proper TimePoint object."""
-    return _point_parse(point_string)
+    return _point_parse(point_string, WorkflowSpecifics.DUMP_FORMAT)
 
 
 @lru_cache(10000)
-def _point_parse(point_string):
+def _point_parse(point_string, _dump_fmt):
```

Comment:
`_dump_fmt` arg does nothing?

Reply:
It is needed for the caching to be valid, seeing as we rely on the `WorkflowSpecifics.DUMP_FORMAT` inside the function. This is how it is done in isodatetime too: https://github.com/metomi/isodatetime/blob/abec7c8ff05ab19d06f5eda790422df54ca72a8b/metomi/isodatetime/data.py#L2174-L2184

Reply:
But this argument is not used within the function.

Reply:
Dunno if `_dump_fmt=WorkflowSpecifics.DUMP_FORMAT` would work with this mutable attribute?

Reply:
If my understanding is correct, `dump_fmt=WorkflowSpecifics.DUMP_FORMAT` gets evaluated at time of function definition (when it is still `None`), not at time of call, so that would not be helpful.

IMHO, it's a "private" (leading underscore) argument in a "private" function, so it's an acceptable solution for the time being